node/src/components/network.rs

```diff
@@ -104,10 +106,14 @@ enum ConnectionState {
 }
 
 #[derive(DataSize)]
-pub(crate) struct Network<REv, P> {
+pub struct Network<REv, P> {
```

Comment:
Do we _really_ need this exposed in our API?

Reply:
Yes. The macro does not work for components that are just `pub(crate)` due to a shortcoming in the Rust compiler that requires a workaround.